chutney/engine/src/main/java/com/chutneytesting/engine/domain/execution/strategies/RetryWithTimeOutStrategy.java

```diff
@@ -106,6 +109,28 @@ public Status execute(ScenarioExecution scenarioExecution,
         return st;
     }
 
+    private long evaluateAndConvertToMilliseconds(String expression, StepDataEvaluator evaluator, ScenarioContext scenarioContext, Map<String, Object> localContext) {
+        if (evaluator == null) {
+            return toMilliSeconds(expression);
+        }
+        try {
+            ScenarioContextImpl mergedContext = new ScenarioContextImpl();
+            mergedContext.putAll(scenarioContext);
+            mergedContext.putAll(localContext);
+
+            Object result = evaluator.evaluate(expression, mergedContext);
+            String evaluatedExpression = result != null ? result.toString() : expression;
+            return toMilliSeconds(evaluatedExpression);
+        } catch (EvaluationException e) {
+            throw new RuntimeException("Failed to evaluate expression: " + expression, e);
+        }
```

Comment:
We generally don't catch the `EvaluateException` which is already a runtime exception.
Let it be.